src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -1298,7 +1402,26 @@ where parameterTypes.SequenceEqual(new[] { iincrementalGeneratorType })
             var analyzerConfigOptionsProvider = analyzerOptions.AnalyzerConfigOptionsProvider();
             verifier.True(analyzerConfigOptionsProvider is not null, "Failed to locate AnalyzerConfigOptionsProvider for project");
 
-            var driver = createMethod.Invoke(null, new[] { convertedSourceGenerators, additionalFiles, project.ParseOptions, analyzerConfigOptionsProvider });
+            object? driver;
+
+            // If we cannot specify generator options or if we are not testing incrementality
+            // don't try to track incremental steps.
+            if (generatorDriverOptionsType is null || IncrementalGeneratorTransforms.Count == 0)
+            {
+                driver = createMethod.Invoke(null, new[] { convertedSourceGenerators, additionalFiles, project.ParseOptions, analyzerConfigOptionsProvider });
+            }
+            else
+            {
+                var constructor = generatorDriverOptionsType.GetConstructor(
+                    new[]
+                    {
+                        generatorDriverOptionsType.GetTypeInfo().Assembly.GetType("Microsoft.CodeAnalysis.IncrementalGeneratorOutputKind")!,
+                        typeof(bool),
+                    });
+                var options = constructor!.Invoke(new object[] { 0, true });
```

Comment:
This would work with just `Activator.CreateInstance` right?

Reply:
We can't use `Activator.CreateInstance` here as that doesn't support implicit conversions of integer values to enum types, and we don't have the enum type available. We could use `Convert.ChangeType` with the `0` to the enum type mentioned above so we could use `Activator.CreateInstance`, but I don't know if that's any cleaner than the alternative.